app/views/OrderPage/index.tsx

```diff
@@ -0,0 +1,181 @@
+    mutation SingleOrder ($bookId: Int!, $qty: Int!) {
+        placeSingleOrder(data: {bookId: $bookId, quantity: $qty}) {
+            errors
+            ok
+        result {
+            orderCode
+            id
+            status
+            totalPrice
+            }
+        }
+    }
+`;
+
+const BOOK_DETAIL = gql`
+query BookById ($id: ID!) {
+    book(id: $id) {
+      id
+      title
+      price
+    }
+  }
+`;
+
+type Book = NonNullable<BookByIdQuery['book']>
+
+interface ItemProps {
+    book: Book;
+    quantity: number;
+    changeQuantity: (qty: number) => void;
+}
+
+function OrderItem(props: ItemProps) {
+    const { book, quantity, changeQuantity } = props;
+    const { id, title, price } = book;
+    const handleQuantityChange = (value: number | undefined,
+        name: string,
+        event: React.FormEvent<HTMLInputElement> | undefined) => {
+        if (value) changeQuantity(value);
+        else changeQuantity(1);
+    };
+
+    return (
+        <div className={styles.container} key={id}>
+            <div className={styles.metaData}>
+                <Container
+                    className={styles.details}
+                    heading="Confirm Order"
+                >
+                    <div className={styles.headerDescription}>
+                        <div className={styles.quantity}>
+                            <TextOutput
+                                label="Book Name"
+                                value={title}
+                            />
+                        </div>
+                        <TextOutput
+                            label="Quantity"
+                            valueType="number"
+                        />
+                        <NumberInput
+                            name="quantity"
+                            value={quantity}
+                            onChange={handleQuantityChange}
+                        />
+                        <TextOutput
+                            label="Price (NPR)"
+                            valueType="number"
+                            value={price}
+                        />
+                        <TextOutput
+                            label="Amount (NPR)"
+                            valueType="number"
+                            value={price * quantity}
+                        />
+                    </div>
+                </Container>
+                <div className={styles.orderButton}>
+                    <Button
+                        name={undefined}
+                        onClick={undefined}
+                        variant="secondary"
+                        icons={<BsCashStack />}
+                    >
+                        Cash On Delivery
+                    </Button>
+                </div>
+            </div>
+        </div>
+    );
+}
+
+function OrderPage() {
+    const location = useLocation();
+    const params = new URLSearchParams(location.search);
+    const id = params.get('book');
```

Comment:
Please use hook provided by react router to get params

Reply:
we need to upgrade the react-router-dom to version 6. In this, we have version 5.

Reply:
We don't need to use version 6.

Why are we using search params btw?